### PR TITLE
Added namespaces and update to our coding guidelines.

### PR DESCRIPTION
**Pull request by danielimprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Due our decision to wrap non-Unreal classes in namespaces, this pr introduces these namespaces and updates the coding guidelines
#### Tests
Compile fine in test suite.
#### Documentation
Coding guidelines updated
#### Primary reviewers
@improbable-valentyn @m-samiec @Vatyx

### CODE REVIEW

File: SpatialGDK/Source/Public/Schema/StandardLibrary.h [hunk 1]
```diff
@@ -10,193 +10,193 @@
...
-		Location.X = 100.0 * Coords.Z;
-		Location.Y = 100.0 * Coords.X;
-		Location.Z = 100.0 * Coords.Y;
-
-		return Location;
-	}
-};
+using WriteAclMap = TMap<Worker_ComponentId, WorkerRequirementSet>;
 
+const Worker_ComponentId ENTITY_ACL_COMPONENT_ID = 50;
```
Comment (danielimprobable) [inline SpatialGDK/Source/Public/Schema/StandardLibrary.h]: Do we want to keep these inside the namespace?

File: SpatialGDK/Source/Public/Schema/Component.h [hunk 2]
```diff
@@ -2,9 +2,14 @@
 
 #pragma once
 
-struct SpatialComponent
+namespace improbable
```
Comment (m-samiec) [inline SpatialGDK/Source/Public/Schema/Component.h]: Should we add a comment to coding standards about namespace tabbings?

File: SpatialGDK/Source/Public/Schema/UnrealObjectRef.h [hunk 3]
```diff
@@ -54,15 +57,24 @@ struct UnrealObjectRef
...
 	uint32 Offset;
-	TSchemaOption<FString> Path;
-	TSchemaOption<UnrealObjectRef> Outer;
+	improbable::TSchemaOption<FString> Path;
+	improbable::TSchemaOption<UnrealObjectRef> Outer;
+};
+
+}
+
+template <>
```
Comment (m-samiec) [inline SpatialGDK/Source/Public/Schema/UnrealObjectRef.h]: Comment?

File: SpatialGDK/Source/Public/Utils/SchemaOption.h [hunk 4]
```diff
@@ -2,111 +2,115 @@
...
+#include "Schema/UnrealObjectRef.h"
 
-template <typename T>
-class TSchemaOption
+namespace improbable
 {
-public:
-	TSchemaOption() = default;
-	~TSchemaOption() = default;
+	template <typename T>
```
Comment (m-samiec) [inline SpatialGDK/Source/Public/Utils/SchemaOption.h]: We should be consistent with our namespace tabbing.

File: SpatialGDK/Source/Public/Schema/Component.h (hunk 2 above)
Comment (danielimprobable) [reply SpatialGDK/Source/Public/Schema/Component.h]: Yes I wasn't sure which tabbing we preferred, I think that is a minor detail though compared to having the namespaces at all

Comment (danielimprobable) [reply SpatialGDK/Source/Public/Schema/Component.h]: @improbable-valentyn @joshuahuburn  what do you think?

File: SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp [hunk 5]
```diff
@@ -219,15 +219,15 @@ FNetworkGUID FSpatialNetGUIDCache::GetNetGUIDFromUnrealObjectRef(const UnrealObj
 	return NetGUID;
 }
 
-UnrealObjectRef FSpatialNetGUIDCache::GetUnrealObjectRefFromNetGUID(const FNetworkGUID& NetGUID) const
+improbable::UnrealObjectRef FSpatialNetGUIDCache::GetUnrealObjectRefFromNetGUID(const FNetworkGUID& NetGUID) const
 {
-	const UnrealObjectRef* ObjRef = NetGUIDToUnrealObjectRef.Find(NetGUID);
-	return ObjRef ? (UnrealObjectRef)*ObjRef : SpatialConstants::UNRESOLVED_OBJECT_REF;
+	const improbable::UnrealObjectRef* ObjRef = NetGUIDToUnrealObjectRef.Find(NetGUID);
```
Comment (girayimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: Users will also have to use `UnrealObjectRef` a lot, let's consider leaving this out of the namespacing.

I guess they can `typedef` if needed.

File: SpatialGDK/Source/Public/Schema/Component.h (hunk 2 above)
Comment (joshuahuburn) [reply SpatialGDK/Source/Public/Schema/Component.h]: I think that this current style is fine. It's how we did it at MS.

File: SpatialGDK/Source/Public/Schema/StandardLibrary.h [hunk 6]
```diff
@@ -37,77 +47,16 @@ struct Coordinates
...
-
-	Coordinates Coords;
-};
-
-using WriteAclMap = TMap<Worker_ComponentId, WorkerRequirementSet>;
-
-const Worker_ComponentId ENTITY_ACL_COMPONENT_ID = 50;
-
-struct SpatialEntityAcl : SpatialComponent
+struct EntityAcl : improbable::Component
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Public/Schema/StandardLibrary.h]: If we're already in the namespace of `improbable` do we need the `improbable::Component` in this context?

Comment (joshuahuburn) [reply SpatialGDK/Source/Public/Schema/StandardLibrary.h]: Same applies to the other `Component`s in this class :D

**Pull request merged**